packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -511,6 +512,79 @@ export type Deployment = {
   versions?: DeploymentVersion[];
 };
 
+// If deploymentId is provided tries fetch only that one deployment and finishes early when found
+async function fetchDeployments(cloudProvider: CloudProvider['type'], deploymentId?: string) {
+  const deployments: Deployment[] = [];
+
+  const bucketName = await cloudProviderLib[cloudProvider].getAirnodeBucket();
+  if (!bucketName) {
+    logger.debug(`No deployments available on ${cloudProvider.toUpperCase()}. Skipping.`);
+    return deployments;
+  }
+
+  const directoryStructure = await cloudProviderLib[cloudProvider].getBucketDirectoryStructure(bucketName);
+  for (const [airnodeAddress, addressDirectory] of Object.entries(directoryStructure)) {
+    if (addressDirectory.type !== FileSystemType.Directory) {
+      logger.warn(
+        `Invalid item in bucket '${bucketName}' (${cloudProvider.toUpperCase()}) with key '${
+          addressDirectory.bucketKey
+        }'. Skipping.`
+      );
+      continue;
+    }
+
+    for (const [stage, stageDirectory] of Object.entries(addressDirectory.children)) {
+      if (stageDirectory.type !== FileSystemType.Directory) {
+        logger.warn(
+          `Invalid item in bucket '${bucketName}' (${cloudProvider.toUpperCase()}) with key '${
+            stageDirectory.bucketKey
+          }'. Skipping.`
+        );
+        continue;
+      }
+
+      const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
+      const bucketConfigPath = `${airnodeAddress}/${stage}/${latestDeployment}/config.json`;
+      const config = JSON.parse(
```

Comment:
I'd wrap this in `go` since this can fail in many ways (users manually edit aws config, we change the region and nodeVersion field). It would be nice to handle this gracefully.

Reply:
I'll do that as part of https://github.com/api3dao/airnode/issues/1449